clients/fides-js/src/lib/tcf.ts

```diff
@@ -5,18 +5,48 @@
  * In the future, this should hold interactions with the CMP API as well as string generation.
  */
 
-import { CmpApi } from "@iabtechlabtcf/cmpapi";
+import { CmpApi, TCData } from "@iabtechlabtcf/cmpapi";
 import { TCModel, TCString, GVL } from "@iabtechlabtcf/core";
 import { makeStub } from "./tcf/stub";
 
 import { EnabledIds } from "./tcf/types";
-import { vendorIsGvl } from "./tcf/vendors";
+import { decodeVendorId, vendorIsAc, vendorIsGvl } from "./tcf/vendors";
 import { PrivacyExperience } from "./consent-types";
+import { FIDES_SEPARATOR } from "./tcf/constants";
+import { FidesEvent } from "./events";
 
+// TCF
 const CMP_ID = 407;
 const CMP_VERSION = 1;
 const FORBIDDEN_LEGITIMATE_INTEREST_PURPOSE_IDS = [1, 3, 4, 5, 6];
 
+// AC
+const AC_SPECIFICATION_VERSION = 1;
+
+/**
+ * Generate an AC String based on TCF-related info from privacy experience.
+ */
+const generateAcString = ({
+  tcStringPreferences,
+}: {
+  tcStringPreferences: Pick<EnabledIds, "vendorsConsent" | "vendorsLegint">;
+}) => {
+  const uniqueIds = Array.from(
+    new Set(
+      [
+        ...tcStringPreferences.vendorsConsent,
+        ...tcStringPreferences.vendorsLegint,
+      ]
+        .filter((id) => vendorIsAc(id))
+        // Convert ac.42 --> 42
+        .map((id) => decodeVendorId(id).id)
+    )
+  );
+  const vendorIds = uniqueIds.join(".");
```

Comment:
do VendorIds need to be sorted ascending? I don't see this specifically in the spec, but see that they're sorted in examples.

Reply:
I'll sort, doesn't hurt!